apps/myaccount/src/components/multi-factor-authentication/multi-factor-authentication.tsx

```diff
@@ -198,27 +198,35 @@ export const MultiFactorAuthentication: React.FunctionComponent<MfaProps> = (pro
                                 isSuperTenantLogin={ isSuperTenantLogin() }
                                 onEnabledAuthenticatorsUpdated={ handleEnabledAuthenticatorsUpdated }
                                 triggerBackupCodesFlow={ () => setInitBackupCodeFlow(true) }
-                                handleSessionTerminationModalVisibility={ 
-                                    () => setShowSessionTerminationModal(true) 
-                                }
+                                handleSessionTerminationModalVisibility={ () => {
+                                    // Show the session termination modal only if the backup code flow is not activated
+                                    // to stop showing duplicate modals.
+                                    if (isSuperTenantLogin() && isTOTPEnabled && isBackupCodesConfigured) {
```

Comment:
We should show the modal if we have generated new backup codes or regenerated them. Once the user closes the backup code modal we need to show the session termination modal.